fairing/deployers/job/job.py

```diff
@@ -25,7 +25,8 @@ class Job(DeployerInterface):
 
     def __init__(self, namespace=None, runs=1, output=None,
                  cleanup=True, labels=None, job_name=DEFAULT_JOB_NAME,
-                 stream_log=True, deployer_type=DEPLOPYER_TYPE):
+                 stream_log=True, deployer_type=DEPLOPYER_TYPE,
+                 mount_credentials=False):
```

Comment:
WDYT about this being not so GCP specific?

How about a list of functions that mutates the pod spec? One such function could be `add_credentials_to_pod_spec` (rename it to `add_gcp_credentials`

So the API would look like

```
def __init__(self, ...., pod_spec_mutators=[add_gcp_credentials])
```

And then the pod spec would be ran through the mutators in order

```
for each fn in pod_mutators:
   fn(self.pod_spec)
```

The interface for the pod spec mutator would just be to accept a pod_spec and return a mutated pod spec (or return a newly modified one)

Reply:
This will have the benefit of 

1. making it easier to plug in different clouds
2. making it easier to modify the pod spec in different ways (we'll most likely have to add more volume mounts, environment variables, and secrets in the future)

Reply:
Thanks for the suggestion! I updated the implementation to use the `pod_spec_mutators` approach, with the deployers taking in a list of mutators accordingly. 

My only concern with this is that it's become more complex from the user-facing point of view. In particular, they need to do something like below:

```
fairing.config.set_deployer(
    'job',
    pod_spec_mutators=[fairing.cloud.gcp.add_gcp_credentials],
    namespace='kubeflow'
)
```